File: modules/swagger-jaxrs/src/main/scala/com/wordnik/swagger/jaxrs/config/DefaultJaxrsConfig.scala
package com.wordnik.swagger.jaxrs.config


import com.wordnik.swagger.config.{ ConfigFactory, ScannerFactory }
import com.wordnik.swagger.jaxrs.reader._
import com.wordnik.swagger.reader._

import javax.servlet.ServletConfig
import javax.servlet.http.HttpServlet

class DefaultJaxrsConfig extends HttpServlet {
  override def init(servletConfig: ServletConfig) = {
    super.init(servletConfig)

    implicit val config = servletConfig
    ConfigFactory.config = new WebXMLReader()
    ScannerFactory.scanner = Some(new DefaultJaxrsScanner())
    servletConfig.getInitParameter("scan.all.resources") match {
      case "true" => {
        val reader = new BasicJaxrsReader
        Option(servletConfig.getInitParameter("ignore.routes")) match {
          case Some(e) => reader.ignoredRoutes = e.split(",").toSet
          case _ =>
        }
        ClassReaders.reader = Some(reader)
      }
      case _ => ClassReaders.reader = Some(new DefaultJaxrsApiReader)
    }
  }
}

File: modules/swagger-jaxrs/src/main/java/com/wordnik/swagger/jaxrs/config/DefaultJaxrsConfig.java
package com.wordnik.swagger.jaxrs.config;

import javax.ws.rs.core.*;

import javax.servlet.ServletConfig;
import javax.servlet.http.HttpServlet;

public class DefaultJaxrsConfig extends HttpServlet {
  @Override
  public void init(ServletConfig servletConfig) throws javax.servlet.ServletException {
    super.init(servletConfig);

    servletConfig.getServletContext().setAttribute("reader", new WebXMLReader(servletConfig));
    servletConfig.getServletContext().setAttribute("scanner", new DefaultJaxrsScanner());
    ReaderConfigUtils.initReaderConfig(servletConfig);
  }
}
